cmd/goplugin-terraform/resource/configuration.go

```diff
@@ -0,0 +1,136 @@
+package resource
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-hclog"
+	"github.com/lyraproj/pcore/px"
+	"github.com/lyraproj/pcore/serialization"
+	"github.com/lyraproj/pcore/types"
+)
+
+// Config is a terraform configuration identified by a working directory
+type Config struct {
+	WorkingDir string
+	UniqueID   *string //HACK to force Read to return a new value, in order to always trigger an Update
+	Output     *px.Value
+}
+
+// ConfigHandler is used to perform CRUD operations on a Config resource
+type ConfigHandler struct{}
+
+// Create a new set of resources
+func (*ConfigHandler) Create(desiredState *Config) (*Config, string, error) {
+	hclog.Default().Debug("Creating Config", "desiredState", desiredState)
+
+	apply(desiredState)
+
+	return desiredState, extID(*desiredState), nil
+}
+
+// Read an existing Config
+func (*ConfigHandler) Read(externalID string) (*Config, error) {
+	hclog.Default().Debug("Reading Config", "externalID", externalID)
+
+	s := uniqueString()
```

Comment:
This seems wrong to me since it will return an object even when the `WorkingDir` doesn't exist. Wouldn't it be better if the `externalID` actually appointed the `WorkingDir` ? That way, this method could check if that directory exists and return the `serviceapi.NotFound()` error if it doesn't. If it is found, then the `Config` can be returned with the `WorkingDir` correctly initialized (from the `externalID`). It will still always trigger an update since it also contains the `UniqueID`.

Reply:
Changed to check path exists and is directory.